### ISSUE TITLE
Remove unnecessary loading of validation data to memory

### ISSUE THREAD

**Issue by angeloskath** [idiap/importance-sampling#23]
In ImportanceTraining classes we unnecessarily load all the data in memory.

https://github.com/idiap/importance-sampling/blob/991230a86d6d3d3a1d8b1a84716ae0db92248e26/importance_sampling/training.py#L288

**Comment by MasterScrat** [comment]
This is problematic - it makes validation completely impossible in some cases, eg when using generators that load heavy images! Is there any way to get around this?

**Comment by angeloskath** [comment]
I agree it is problematic and it needs to be changed. The simplest solution I can think of is to do an external for loop and evaluate the unwrapped model. Something like the following:

```python
model = .....
wrapped = ImportanceTraining(model)
for i in range(epochs):
    wrapped.fit(...)
    model.evaluate(...)
```
